### PR TITLE
Add Kyber Internal Governance for new KyberDAO

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

**Review by Anyhowclick — DISMISSED** [review]
LGTM

File: contracts/misc/KyberInternalGovernance.sol [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            cycle, index, address(this), tokens, cumulativeAmounts, merkleProof
+        );
+
+        for(uint256 i = 0; i < tokens.length; i++) {
+            uint256 bal = tokens[i] == ETH_TOKEN_ADDRESS ?
+                address(this).balance : tokens[i].balanceOf(address(this));
+            if (bal > 0) _transferToken(tokens[i], bal);
+        }
+        
+        emit ClaimedRewards(cycle, index, tokens, claimAmounts);
```
Comment (ducquangkstn) [inline contracts/misc/KyberInternalGovernance.sol:91]: this event is duplicated in RewardDistributor
and there is the case when another people call claim reward for this contract

File: contracts/misc/KyberInternalGovernance.sol [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import {PermissionAdmin} from '@kyber.network/utils-sc/contracts/PermissionAdmin.sol';
+import {PermissionOperators} from '@kyber.network/utils-sc/contracts/PermissionOperators.sol';
+import {Utils} from '@kyber.network/utils-sc/contracts/Utils.sol';
+import {IERC20Ext} from '@kyber.network/utils-sc/contracts/IERC20Ext.sol';
+import {IKyberGovernance} from '../interfaces/governance/IKyberGovernance.sol';
+import {IRewardsDistributor} from '../interfaces/rewardDistribution/IRewardsDistributor.sol';
+
+
+/**
+    Internal contracts to participate in KyberDAO and claim rewards for Kyber
```
Comment (ducquangkstn) [inline contracts/misc/KyberInternalGovernance.sol:15]: Pls add a comment that ppl should delegate to this contract (no deposit required).
This contract will vote on behalf of them and transfer tokens to the recipient

File: contracts/misc/KyberInternalGovernance.sol [hunk 3]
```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.7.6;
+
+
+import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import {PermissionAdmin} from '@kyber.network/utils-sc/contracts/PermissionAdmin.sol';
```
Comment (ducquangkstn) [inline contracts/misc/KyberInternalGovernance.sol:6]: `PermissionOperators` extends from `PermissionAdmin` so this import is redundant

**Review by ducquangkstn — COMMENTED** [review]
LGTM

However, this should not be misc folder. May be in the same folder with PoolMaster?

File: contracts/misc/KyberInternalGovernance.sol (hunk 3 above)
Comment (manhlx3006) [reply contracts/misc/KyberInternalGovernance.sol:6]: actually you'll need to import both contracts

File: contracts/misc/KyberInternalGovernance.sol (hunk 2 above)
Comment (manhlx3006) [reply contracts/misc/KyberInternalGovernance.sol:15]: Added

**Review by ducquangkstn — DISMISSED** [review]

**Review by ducquangkstn — COMMENTED** [review]
Why there are 2 binary files on this PR

**Review by ducquangkstn — APPROVED** [review]

File: contracts/misc/KyberInternalGovernance.sol [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import {PermissionOperators} from '@kyber.network/utils-sc/contracts/PermissionOperators.sol';
```
Comment (Anyhowclick) [inline contracts/misc/KyberInternalGovernance.sol:7]: can do `import {PermissionAdmin, PermissionOperators} from '@kyber.network/utils-sc/contracts/PermissionOperators.sol';` for the reason Quang stated. See pool master contract haha

**Review by Anyhowclick — CHANGES_REQUESTED** [review]

File: contracts/misc/KyberInternalGovernance.sol (hunk 4 above)
Comment (manhlx3006) [reply contracts/misc/KyberInternalGovernance.sol:7]: code has been deployed, so maybe can just keep this way ...

**Review by Anyhowclick — APPROVED** [review]

**Pull request merged**